src/zenml/integrations/gcp/model_registries/vertex_model_registry.py

```diff
@@ -0,0 +1,309 @@
+                display_name=f"{name}_{version}",
+                serving_container_image_uri=serving_container_image_uri,
+                description=description,
+                is_default_version=is_default_version,
+                labels=metadata_dict,
+            )
+            return RegistryModelVersion(
+                version=version_info.version_id,
+                model_source_uri=version_info.resource_name,
+                model_format="Custom",  # Vertex AI doesn't provide this info directly
+                registered_model=self.get_model(version_info.name),
+                description=description,
+                created_at=version_info.create_time,
+                last_updated_at=version_info.update_time,
+                stage=ModelVersionStage.NONE,  # Vertex AI doesn't have built-in stages
+                metadata=metadata,
+            )
+        except Exception as e:
+            raise RuntimeError(f"Failed to register model version: {str(e)}")
+
+    def delete_model_version(
+        self,
+        name: str,
+        version: str,
+    ) -> None:
+        """Delete a model version from the Vertex AI model registry."""
+        self.setup_aiplatform()
+        try:
+            model_version = aiplatform.ModelVersion(
+                model_name=f"{name}@{version}"
+            )
+            model_version.delete()
+        except Exception as e:
+            raise RuntimeError(f"Failed to delete model version: {str(e)}")
+
+    def update_model_version(
+        self,
+        name: str,
+        version: str,
+        description: Optional[str] = None,
+        metadata: Optional[ModelRegistryModelMetadata] = None,
+        remove_metadata: Optional[List[str]] = None,
+        stage: Optional[ModelVersionStage] = None,
+    ) -> RegistryModelVersion:
+        """Update a model version in the Vertex AI model registry."""
+        self.setup_aiplatform()
+        try:
+            model_version = aiplatform.Model(model_name=f"{name}@{version}")
+            labels = model_version.labels
+            if metadata:
+                metadata_dict = metadata.model_dump() if metadata else {}
+                for key, value in metadata_dict.items():
+                    labels[key] = value
+            if remove_metadata:
+                for key in remove_metadata:
+                    labels.pop(key, None)
+            model_version.update(description=description, labels=labels)
+            return self.get_model_version(name, version)
+        except Exception as e:
+            raise RuntimeError(f"Failed to update model version: {str(e)}")
+
+    def get_model_version(
+        self, name: str, version: str
+    ) -> RegistryModelVersion:
+        """Get a model version from the Vertex AI model registry."""
+        self.setup_aiplatform()
+        try:
+            model_version = aiplatform.Model(model_name=f"{name}@{version}")
+            return RegistryModelVersion(
```

Comment:
converting a vertex model version to `RegistryModelVersion` happens a lot of times, can we maybe put that in a helper method?